### PR TITLE
Docs for resource loading upgrade path

### PR DESCRIPTION
**Pull request by chillu**
See https://github.com/silverstripe/silverstripe-framework/pull/7065

### CODE REVIEW

File: docs/en/04_Changelogs/4.0.0.md [hunk 1]
```diff
@@ -108,6 +109,65 @@ Core template locations have moved - if you're including or overriding these
...
+// before
+Requirements::css('framework/css/styles.css');
+
+// after
+Requirements::css('silverstripe/framework: css/styles.css');
+```
+
+Usage with custom logic:
+
+```php
```
Comment (chillu) [inline docs/en/04_Changelogs/4.0.0.md]: I really wish we could use `diff` syntax here, but the underlying syntax highlighter used for docs.ss.org doesn't support it...

File: docs/en/04_Changelogs/4.0.0.md [hunk 2]
```diff
@@ -108,6 +109,65 @@ Core template locations have moved - if you're including or overriding these
...
+$baseFilePath = BASE_PATH . '/composer.json'; 
+$mysiteFilePath = 'mysite/css/styles.css';
+$themesFilePath = SSViewer::get_theme_folder() . '/css/styles.css';
+$themeFolderPath = THEMES_DIR . '/simple';
+
+// after
+use SilverStripe\Core\Manifest\ModuleLoader;
+use SilverStripe\View\ThemeResourceLoader;
+$moduleFilePath = ModuleLoader::getModule('silverstripe/framework')->getRelativeResourcePath('MyFile.php');
+$baseFilePath = Director::baseFolder() . '/composer.json';
```
Comment (chillu) [inline docs/en/04_Changelogs/4.0.0.md]: Not quite sure if this is the right way to do this now? It sure beats `BASE_PATH`, but it's still static.

File: docs/en/04_Changelogs/4.0.0.md (hunk 1 above)
Comment (robbieaverill) [inline docs/en/04_Changelogs/4.0.0.md]: Bah - does parsedown not support diff?

Comment (chillu) [inline docs/en/04_Changelogs/4.0.0.md]: Parsedown doesn't do syntax highlighting? We're using https://github.com/google/code-prettify as a JS lib, and that doesn't do diff: https://github.com/google/code-prettify/issues/62. Ideas?

Comment (robbieaverill) [inline docs/en/04_Changelogs/4.0.0.md]: Doh, of course. I feel like we have had this chat before, would need a separate JS lib just for diff highlighting

Comment (robbieaverill) [inline docs/en/04_Changelogs/4.0.0.md]: [highlight.js](https://highlightjs.org) supports a large range of syntaxes, including diff - perhaps we could switch?

Comment (chillu) [inline docs/en/04_Changelogs/4.0.0.md]: https://github.com/silverstripe/silverstripe-docsviewer/pull/134 :)

File: docs/en/04_Changelogs/4.0.0.md [hunk 3]
```diff
@@ -108,6 +109,65 @@ Core template locations have moved - if you're including or overriding these
...
+$mysiteFilePath = ModuleLoader::getModule('mysite')->getRelativeResourcePath('css/styles.css');
+$themesFilePath = ThemeResourceLoader::inst()->findThemedResource('css/styles.css');
+$themeFolderPath = ThemeResourceLoader::inst()->getPath('simple');
+```
+
+To ensure consistency, we've also removed support for path constants:
```
Comment (tractorcow) [inline docs/en/04_Changelogs/4.0.0.md]: We haven't removed these.

Comment (chillu) [inline docs/en/04_Changelogs/4.0.0.md]: Replaced "removed" with "deprecated"

File: docs/en/04_Changelogs/4.0.0.md [hunk 4]
```diff
@@ -108,6 +109,65 @@ Core template locations have moved - if you're including or overriding these
...
+$themeFolderPath = ThemeResourceLoader::inst()->getPath('simple');
+```
+
+To ensure consistency, we've also deprecated support for path constants:
+
+ * `FRAMEWORK_DIR` and `FRAMEWORK_PATH`
+ * `FRAMEWORK_ADMIN_DIR` and `FRAMEWORK_ADMIN_PATH`
+ * `FRAMEWORK_ADMIN_THIRDPARTY_DIR` and `FRAMEWORK_ADMIN_THIRDPARTY_PATH`
+ * `THIRDPARTY_DIR` and `THIRDPARTY_PATH`
+ * `ASSETS_DIR` and `ASSETS_PATH`
```
Comment (tractorcow) [inline docs/en/04_Changelogs/4.0.0.md]: I think we have a "`ASSETS_PATH`" env lookup in config f or assets store configuration, but we could still deprecate it I think.

Comment (chillu) [inline docs/en/04_Changelogs/4.0.0.md]: You're right, it's too early to deprecate that - there's no appropriate getter. Which we'll need to solve, but it's out of scope of this docs-focused PR

**Pull request merged**